# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/l10n/tests/mochitest/dom_localization/test_overlay.html
# Commit: 626b40e3aad1
# Full Hash: 626b40e3aad147d59a439b0bd1d5dbd328ad6b1a
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-02 21:43:56
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 1cbab987bb7d -r 626b40e3aad1 dom/l10n/tests/mochitest/dom_localization/test_overlay.html
--- a/dom/l10n/tests/mochitest/dom_localization/test_overlay.html	Mon Aug 02 09:54:10 2021 +0000
+++ b/dom/l10n/tests/mochitest/dom_localization/test_overlay.html	Mon Aug 02 09:54:10 2021 +0000
@@ -7,20 +7,24 @@
   <link rel="stylesheet" type="text/css" href="chrome://mochikit/content/tests/SimpleTest/test.css">
   <script type="application/javascript">
   "use strict";
-  async function* generateBundles(resourceIds) {
-    const bundle = new FluentBundle("en-US");
-    bundle.addResource(new FluentResource("title = <strong>Hello</strong> World"));
-    bundle.addResource(new FluentResource(`title2 = This is <a data-l10n-name="link">a link</a>!`));
-    yield bundle;
-  }
+  const l10nReg = new L10nRegistry();
+  const fs = [
+    { path: "/localization/en-US/mock.ftl", source: `
+title = <strong>Hello</strong> World
+title2 = This is <a data-l10n-name="link">a link</a>!
+` },
+  ];
+  const source = L10nFileSource.createMock("test", ["en-US"], "/localization/{locale}", fs);
+  l10nReg.registerSources([source]);
 
   window.onload = async function() {
     SimpleTest.waitForExplicitFinish();
 
     const domLoc = new DOMLocalization(
-      [],
+      ["/mock.ftl"],
       false,
-      { generateBundles },
+      l10nReg,
+      ["en-US"],
     );
 
     const p1 = document.querySelectorAll("p")[0];